controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -256,15 +256,29 @@ func (a *Adapter) determineIfAllIntegrationPipelinesPassed(integrationPipelineRu
 	allIntegrationPipelineRunsPassed := true
 	for _, integrationPipelineRun := range *integrationPipelineRuns {
 		integrationPipelineRun := integrationPipelineRun // G601
-		pipelineRunOutcome, err := h.CalculateIntegrationPipelineRunOutcome(a.client, a.context, a.logger.Logger, &integrationPipelineRun)
+		if !h.HasPipelineRunFinished(&integrationPipelineRun) {
+			a.logger.Info("The integration PipelineRun has not finished yet")
+			allIntegrationPipelineRunsPassed = false
+			continue
+		}
+		pipelineRunOutcome, err := h.GetIntegrationPipelineRunOutcome(a.client, a.context, &integrationPipelineRun)
 		if err != nil {
 			a.logger.Error(err, "Failed to get Integration PipelineRun outcome")
 			return false, err
 		}
-		if !pipelineRunOutcome {
-			a.logger.Info("Integration PipelineRun did not pass all tests")
+		if !pipelineRunOutcome.GetOutcome() {
+			if !pipelineRunOutcome.HasPipelineRunSucceeded() {
+				a.logger.Error(
+					fmt.Errorf("integration PipelineRun failed: %s", h.GetPipelineRunFailedReason(&integrationPipelineRun)),
+					"PipelineRun failed without test results of TaskRuns",
+				)
+
+			} else {
+				a.logger.Info("Integration PipelineRun did not pass all tests")
+			}
 			allIntegrationPipelineRunsPassed = false
```

Comment:
Asking for a friend, why do we need to finish this loop when we know that result is already false?

k8s api has lot of fun even without us querying for tasks results we don't need

Reply:
I am curious too. Maybe it is to find and log the error, otherwise I feel we don't need to finish the loop if the result is already false.

Reply:
Yes, it was left this way to help with debugging all pipelineRuns, but I agree that we can finish the loop at this time to stop wasting resources. Both developers and users will have more insight into what happened with their integration tests when the Snapshots start getting exposed in the UI.

Reply:
I can do this change in separate PR, I don't want to extend scope of this even more